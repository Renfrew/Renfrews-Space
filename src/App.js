import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

import CssBaseline from '@material-ui/core/CssBaseline';

import Home from './features/Home';
import NotFound404 from './features/NotFound404';

const routes = [
  { path: '/', component: <Home /> },
  { path: '*', component: <NotFound404 /> },
];

function App() {
  return (
    <React.Fragment>
      <CssBaseline />
      <Router>
        <Switch>
          {routes.map((el) => (
            <Route key={el} path={el.path}>
              {el.component}
            </Route>
          ))}
        </Switch>
      </Router>
    </React.Fragment>
  );
}

export default App;
