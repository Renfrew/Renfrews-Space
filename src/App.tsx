import React, { ReactElement } from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

import CssBaseline from '@material-ui/core/CssBaseline';

// Temporary until Material-UI has a stable version
import StyledEngineProvider from '@material-ui/core/StyledEngineProvider';

import Home from './features/Home';
import Navigation from './features/Navigation';

const routes = [
  { path: '/:path', component: <Navigation /> },
  { path: '/', component: <Home /> },
];

function App(): ReactElement {
  return (
    <StyledEngineProvider injectFirst>
      <CssBaseline />
      <Router>
        <Switch>
          {routes.map((el) => (
            <Route key={el.path} path={el.path}>
              {el.component}
            </Route>
          ))}
        </Switch>
      </Router>
    </StyledEngineProvider>
  );
}

export default App;
