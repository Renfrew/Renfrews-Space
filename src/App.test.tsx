import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

describe('Testing the Home page', () => {
  it('The home page renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
