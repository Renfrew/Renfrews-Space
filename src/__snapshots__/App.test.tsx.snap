// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Home page The home page renders correctly 1`] = `
<main
  className="MuiContainer-root MuiContainer-disableGutters makeStyles-container-1 css-q7bdj4-MuiContainer-root"
/>
`;
